lib/main.js

```diff
@@ -47,6 +47,30 @@ const Hydrogen = {
 
     this.markerBubbleMap = {};
 
+    let skipLanguageMappingsChange = false;
+    store.subscriptions.add(
+      atom.config.onDidChange(
+        "Hydrogen.languageMappings",
+        ({ newValue, oldValue }) => {
+          if (skipLanguageMappingsChange) {
+            skipLanguageMappingsChange = false;
+            return;
+          }
+
+          if (store.runningKernels.size != 0) {
+            skipLanguageMappingsChange = true;
+
+            atom.config.set("Hydrogen.languageMappings", oldValue);
```

Comment:
Yeah, I spoke to fast, we would probably need to call it in a `onWillChange` handler but `atom.config` doesn't has one.